### PR TITLE
Make sure that Either CPS on Scala 3 handle Lefts correctly

### PR DESCRIPTION
**Pull request by romanowski**

### CODE REVIEW

File: modules/build-macros/src/test/scala/scala/build/CPSTest.scala [hunk 1]
```diff
@@ -50,6 +53,14 @@ class CPSTest extends munit.FunSuite {
     new V
   }
 
+  // When changing something in EitherCPS.scala uncomment this and make sure it does not compile
```
Comment (lwronski) [inline modules/build-macros/src/test/scala/scala/build/CPSTest.scala]: Maybe, we can use scala-cli to test such cases.

**Review by lwronski — APPROVED** [review]

File: modules/build-macros/src/test/scala/scala/build/CPSTest.scala [hunk 2]
```diff
@@ -50,6 +53,14 @@ class CPSTest extends munit.FunSuite {
...
+  // def ee3: Either[E2, V] = either {
```
Comment (alexarchambault) [inline modules/build-macros/src/test/scala/scala/build/CPSTest.scala]: In Scala 2, we could use [`shapeless.test.illTyped`](https://github.com/milessabin/shapeless/blob/19a9c3a5951e4197084285bd568c2fef6717ef81/core/src/main/scala/shapeless/test/typechecking.scala#L33) to test that this doesn't compile.

File: modules/build-macros/src/main/scala-2.13/scala/build/EitherCps.scala [hunk 3]
```diff
@@ -10,7 +10,7 @@ object EitherCps {
   final case class Helper[E]() {
```
Comment (alexarchambault) [inline modules/build-macros/src/main/scala-2.13/scala/build/EitherCps.scala:10]: Maybe we could remove the `case` here? (seems it was superfluous, and that would justify the use of `new` below)

**Comment by alexarchambault** [comment]
To be safer, we should check that the output in negative tests prints an error we expect, IMO.

File: modules/build-macros/src/test/scala/scala/build/CPSTest.scala (hunk 2 above)
Comment (romanowski) [reply modules/build-macros/src/test/scala/scala/build/CPSTest.scala]: The problem is that it compiles on Scala 2

Comment (alexarchambault) [reply modules/build-macros/src/test/scala/scala/build/CPSTest.scala]: Oh, ok.

**Review by alexarchambault — APPROVED** [review]
Approved, but squash-and-merge, please :)

**Pull request merged**